data-cube-curation/templates/deployment.yaml

```diff
@@ -0,0 +1,172 @@
+{{- $fullName := include "data-cube-curation.fullname" . -}}
+{{- $sparqlSecretName := include "data-cube-curation.sparql.secretName" . -}}
+{{- $gitlabSecretName := include "data-cube-curation.gitlab.secretName" . -}}
+{{- $s3SecretName := include "data-cube-curation.s3.secretName" . -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $fullName }}
+  labels:
+    {{- include "data-cube-curation.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "data-cube-curation.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "data-cube-curation.selectorLabels" . | nindent 8 }}
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 3000
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
+              port: http
+            initialDelaySeconds: 30
+          readinessProbe:
+            httpGet:
+              path: /
+              port: http
+            initialDelaySeconds: 30
```

Comment:
This should be shorter (e.g. 3s) I suggest, otherwise it will take at least 30s for new pods to become ready (receive traffic).

Reply:
Considering that the `periodSeconds` defaults to 10 and the `failureThreshold` to 3 (see [reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#probe-v1-core)), the container can take up to 30s before getting killed without the initial delay. Not sure if adding 3s to that makes much sense.

Reply:
Yes agree, for the readiness probe we can get rid of the delay

Reply:
Done